samples/net/connectivity_check/src/main.c

```diff
@@ -0,0 +1,146 @@
+static K_WORK_DELAYABLE_DEFINE(dwork, trigger_check);
+static K_SEM_DEFINE(quit_lock, 0, K_SEM_MAX_LIMIT);
+static struct net_if *my_iface;
+static bool connected;
+
+#define TRIGGER_TIMEOUT 5
+static k_timeout_t trigger_timeout = K_SECONDS(TRIGGER_TIMEOUT);
+
+void quit(void)
+{
+	k_sem_give(&quit_lock);
+}
+
+static void trigger_check(struct k_work *work)
+{
+	static int count;
+
+	ARG_UNUSED(work);
+
+	if (count >= CONFIG_NET_SAMPLE_ONLINE_COUNT) {
+		quit();
+	}
+
+	count++;
+}
+
+#define EVENTS (NET_EVENT_CONNECTIVITY_ONLINE | NET_EVENT_CONNECTIVITY_OFFLINE)
+
+static void event_handler(uint32_t mgmt_event, struct net_if *iface,
+			  void *info, size_t info_length, void *user_data)
+{
+	if ((mgmt_event & EVENTS) != mgmt_event) {
+		return;
+	}
+
+	if (mgmt_event == NET_EVENT_CONNECTIVITY_ONLINE) {
+		char ifname[CONFIG_NET_INTERFACE_NAME_LEN];
+
+		(void)net_if_get_name(iface, ifname, sizeof(ifname));
+
+		LOG_INF("Connected online on interface %s (%p)",
+			ifname, iface);
+		connected = true;
+		my_iface = iface;
+
+		/* Wait a bit, and then trigger online check again */
+		k_work_reschedule(&dwork, trigger_timeout);
+		return;
+	}
+
+	if (mgmt_event == NET_EVENT_CONNECTIVITY_OFFLINE) {
+		if (connected == false) {
+			LOG_INF("Waiting network to get online.");
+		} else {
+			LOG_INF("Network no longer online");
+			connected = false;
+		}
+
+		return;
+	}
+}
+
+static NET_MGMT_REGISTER_EVENT_HANDLER(connectivity_events, EVENTS,
+				       event_handler, NULL);
+
+static int tls_setup_cb(struct net_if *iface,
+			const sec_tag_t **sec_tag_list,
+			size_t *sec_tag_size,
+			const char *url,
+			const char *host,
+			const char *port,
+			struct sockaddr *dst,
+			void *user_data)
+{
+	if (!IS_ENABLED(CONFIG_NET_SOCKETS_SOCKOPT_TLS)) {
+		return -ENOTSUP;
+	}
+
+	/* In this test app, we are only interested in the security tags */
+	ARG_UNUSED(iface);
+	ARG_UNUSED(url);
+	ARG_UNUSED(host);
+	ARG_UNUSED(port);
+	ARG_UNUSED(dst);
+	ARG_UNUSED(user_data);
+
+#if defined(CONFIG_NET_SOCKETS_SOCKOPT_TLS)
+	static sec_tag_t my_sec_tag_list[] = {
+		CA_CERTIFICATE_TAG,
+	};
+
+	int ret = tls_credential_add(CA_CERTIFICATE_TAG,
+				     TLS_CREDENTIAL_CA_CERTIFICATE,
+				     ca_certificate,
+				     sizeof(ca_certificate));
+	if (ret < 0 && ret != -EEXIST) {
+		LOG_ERR("Failed to register public certificate: %d", ret);
+	}
+
+	*sec_tag_list = my_sec_tag_list;
+	*sec_tag_size = sizeof(my_sec_tag_list);
+#endif
+
+	return 0;
+}
+
+int main(void)
+{
+	/* Default strategy is PING so only set the HTTP one if enabled. */
+	if (IS_ENABLED(CONFIG_NET_CONNECTION_MANAGER_ONLINE_CHECK_HTTP)) {
+		conn_mgr_set_online_check_strategy(NET_CONN_MGR_ONLINE_CHECK_HTTP);
+	}
+
+	if (IS_ENABLED(CONFIG_NET_SOCKETS_SOCKOPT_TLS)) {
+		int ret;
+
+		ret = conn_mgr_register_online_checker_cb(tls_setup_cb, NULL);
+		if (ret < 0) {
+			LOG_WRN("Cannot enable TLS (%d)", ret);
+		}
+	}
+
+	k_sem_take(&quit_lock, K_FOREVER);
+
+	printk("DONE!\n");
```

Comment:
The rest of tha sample code uses log macros, perhaps we could stick to that.

Reply:
The `printk()` was there on purpose so that it is only thing printed on this line. The Docker test script monitors the output and stops executing when the `DONE!` is printed.

Reply:
I added a comment in the code to describe why the print is there.